### PR TITLE
Fail the work item when the azure pipelines upload fails

### PR DESCRIPTION
**Pull request by alexperovich**
### To double check:

* [ ] The right tests are in and and the right validation has happened.  Guidance:  https://github.com/dotnet/core-eng/tree/master/Documentation/Validation

### CODE REVIEW

File: src/Microsoft.DotNet.Helix/Sdk/tools/azure-pipelines/reporter/run.py [hunk 1]
```diff
@@ -105,6 +112,10 @@ def main():
     q.join()
 
     log.info("Main thread exiting")
+    
+    with workerFailedLock:
+        if workerFailed:
+            sys.exit(1337)
```
Comment (MattGal) [inline src/Microsoft.DotNet.Helix/Sdk/tools/azure-pipelines/reporter/run.py:118]: nice

**Review by MattGal — APPROVED** [review]
LGTM

File: src/Microsoft.DotNet.Helix/Sdk/tools/azure-pipelines/reporter/run.py [hunk 2]
```diff
@@ -33,13 +36,16 @@ def __process(self, batch):
...
     def run(self):
+        global workerFailed, workerFailedLock
         self.__print("starting...")
         while True:
             try:
                 item = self.queue.get()
                 self.__process(item)
             except:
                 self.__print("got error: {}".format(traceback.format_exc()))
+                with workerFailedLock:
```
Comment (ChadNedzlek) [inline src/Microsoft.DotNet.Helix/Sdk/tools/azure-pipelines/reporter/run.py:47]: Does it really need the lock to set it to true?

Comment (ChadNedzlek) [reply src/Microsoft.DotNet.Helix/Sdk/tools/azure-pipelines/reporter/run.py:47]: 5 threads all setting it to true simultaneously shouldn't be a problem, unless python has some weird concurrent modification detection logic built in, which would be pretty crazy.

Comment (alexperovich) [reply src/Microsoft.DotNet.Helix/Sdk/tools/azure-pipelines/reporter/run.py:47]: I don't think it hurts anything, and I would rather have it.

Comment (ChadNedzlek) [reply src/Microsoft.DotNet.Helix/Sdk/tools/azure-pipelines/reporter/run.py:47]: It's just needless complication in already complicated code, but if you like it, I don't have any strong objections.  It's just whenever I see anything with locks, I have to engage a LOT more brainpower to attempt to understand what the code is doing, and in this case, it's just wasted time.

Comment (missymessa) [reply src/Microsoft.DotNet.Helix/Sdk/tools/azure-pipelines/reporter/run.py:47]: image

File: src/Microsoft.DotNet.Helix/Sdk/tools/azure-pipelines/reporter/run.py (hunk 1 above)
Comment (MattGal) [reply src/Microsoft.DotNet.Helix/Sdk/tools/azure-pipelines/reporter/run.py:118]: Weird, when it actually happens it's not 1337 :(

**Pull request merged**